### PR TITLE
add filters to audio instance

### LINKED ISSUE #54: Questions about filters and fading
I would like to ask about 2 things:

1. Is there a way to play the same sound with different filters at the same time?
e.g. Panning effect left-to-right for 5 sounds played 100 ms after each other.

2. From what I understand, the library supported fadeIn/fadeOut features. Were they removed?

Thank you.

### LINKED ISSUE #73: Setting filters per-instance
My use case: I'd like to have multiple instances of a single sound playing at once but with different pans. To adjust the pan I'd need to apply the StereoPan filter to the sound. Unfortunately the filters property exists only on the sound and not IMediaInstance so this is not possible.

Could we have support for per-instance filters on top of sound and global filters?

### PR DESCRIPTION
**Pull request by yueyuzhao**
### Resolves

- Resolves #73 
- Resolves #54 

### Proposed Changes

add `filters` to `PlayOptions` and `WebAudioInstance`

### Reason for Changes

Please see the reference issues.

### Test Coverage

- [x] code lint
- [x] add new sample

### CODE REVIEW

File: src/webaudio/WebAudioInstance.ts [hunk 1]
```diff
@@ -157,6 +161,33 @@ class WebAudioInstance extends EventEmitter implements IMediaInstance
...
+                }
+            });
+            this._filters = null;
+            // Reconnect direct path
+            this._source.connect(this._gain);
+        }
+        if (filters && filters.length)
+        {
+            this._filters = filters.slice(0);
+        }
```
Comment (bigtimebuddy) [inline src/webaudio/WebAudioInstance.ts:187]:
```suggestion
        if (this._filters)
        {
            this._filters?.filter(filter => !!filter).forEach(filter => filter.disconnect());
            this._filters = null;
            // Reconnect direct path
            this._source.connect(this._gain);
        }
        this._filters = filters?.length ? [...filters] : null;
```

**Review by bigtimebuddy — COMMENTED** [review]
Nice work, I only have some minor code suggestions.

File: src/webaudio/WebAudioInstance.ts [hunk 2]
```diff
@@ -337,6 +400,14 @@ class WebAudioInstance extends EventEmitter implements IMediaInstance
             this._media.context.events.off('refreshPaused', this.refreshPaused, this);
             this._media = null;
         }
+        if (this._filters)
+        {
+            this._filters.forEach((filter) =>
+            {
+                filter.disconnect();
+            });
+        }
```
[the remark this answers is not in the document]
Comment (yueyuzhao) [reply src/webaudio/WebAudioInstance.ts:409]: Another question here: should we call `filter.destroy()` here? Can filters set to an audio instance be bound to another?

[the remark this answers is not in the document]
Comment (bigtimebuddy) [reply src/webaudio/WebAudioInstance.ts]: The user constructs filters and should also destroy them.

**Review by bigtimebuddy — APPROVED** [review]

**Pull request merged**